qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,259 @@
+        """Default values for the pulse if no schedule is given.
+        Users can set the positive and negative rotation schedules with
+
+        .. code-block::
+
+            drag.set_experiment_options(rp=xp_schedule, rm=xm_schedule)
+        """
+        options = super()._default_experiment_options()
+
+        options.rp = None
+        options.rm = None
+        options.amp = 0.2
+        options.duration = 160
+        options.sigma = 40
+        options.reps = [1, 3, 5]
+        options.betas = np.linspace(-5, 5, 51)
+
+        return options
+
+    # pylint: disable=arguments-differ
+    def set_experiment_options(self, reps: Optional[List] = None, **fields):
+        """Raise if reps has a length different from three.
+
+        Raises:
+            CalibrationError: if the number of repetitions is different from three.
+        """
+
+        if reps is None:
+            reps = [1, 3, 5]
+        else:
+            reps = sorted(reps)  # ensure reps 1 is the lowest frequency.
+
+        if len(reps) != 3:
+            raise CalibrationError(
+                f"{self.__class__.__name__} must use exactly three repetition numbers. "
+                f"Received {reps} with length {len(reps)} != 3."
+            )
+
+        super().set_experiment_options(reps=reps, **fields)
+
+    def __init__(self, qubit: int):
+        """
+        Args:
+            qubit: The qubit for which to run the Drag calibration.
+        """
+
+        super().__init__([qubit])
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the Drag calibration.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            circuits: The circuits that will run the Drag calibration.
+
+        Raises:
+            CalibrationError:
+                - If the beta parameters in the xp and xm pulses are not the same.
+                - If either the xp or xm pulse do not have at least one Drag pulse.
+                - If the number of different repetition series is not three.
+        """
+        # TODO this is temporary logic.
+        self.set_analysis_options(
+            data_processor=get_to_signal_processor(
+                meas_level=self.run_options.meas_level,
+                meas_return=self.run_options.meas_return,
+            ),
+        )
+
+        plus_sched = self.experiment_options.rp
+        minus_sched = self.experiment_options.rm
+
+        if plus_sched is None:
+            beta = Parameter("β")
+            with pulse.build(backend=backend, name="xp") as plus_sched:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
```

Comment:
This doesn't block merging but you can use `pulse.drive_channel` because you have backend in the builder.